### DISCUSSION TITLE
Remove includeRoomsAsLights?

### DISCUSSION THREAD

**Discussion by faanskit** [icanos/hassio-plejd#191] [Show and tell]
With the introduction of [Areas](https://www.home-assistant.io/integrations/config/#areas) in Home Assistant and in particular the possibility for devices to publish their [Suggested Area](https://www.home-assistant.io/blog/2021/03/03/release-20213/#suggested-areas) as released in version 2021.3., the includeRoomsAsLights as functions is maybe outdated and should possibly be deprecated.

In a coming PR, currently only available in a [branch](https://github.com/faanskit/hassio-plejd/tree/feature/room-as-area/plejd) in my fork, I have implemented support for suggested_area.

All rooms configured in the Plejd App will be generated as Areas in Home Assistant, and devices will be connected to these areas. Devices can be moved to other areas in Home Assistant, as this is merely _suggested_ areas.

In Home Assistant, Areas will appear in the Lovelace Ui.
![kontor](https://user-images.githubusercontent.com/10439868/117489090-75591300-af6d-11eb-9667-e8085e37c1b0.JPG)

The rationale for proposing to remove includeRoomsAsLights is simply because the function is now redundant. All devices in a room can be handled directly in the Ui:
![kontor_2](https://user-images.githubusercontent.com/10439868/117489266-b5b89100-af6d-11eb-8bf0-5b33d22ed0a4.JPG)

In fact, this is better since non-plejd devices can be connected to the same bulc on/off function:

Possible drawbacks:
- Likely slower to execute
- Implemented automations may break

But: To remove it will not really make a any difference, so maybe better to just let-it-be

**Comment by SweVictor** [comment]
I've been thinking about the Plejd rooms a bit too, see #152

To me, the implementation is quite lacking today, and I'd be quite happy to remove them code-wise. 

Functionality-wise, Plejd rooms is the only way (I know of) to multi-target lights (light groups). SO - with creative "rooming" you could actually get very quick state changes or transitions for many lights. But it gets quite complicated when lights inside the room has different brightnesses you want to transition from.

Either way - the room hinting over MQTT I think is a very welcome change.

**Reply by faanskit** [reply]
Honestly I don't think Plejd DIM-01/DIM-02 is fit for purpose for the use case you describe (multi-dimming). Transitions in the Plejd ecosystem is configured per load and cannot be controlled dynamically. Any attempt from the "outside" will be jittery and it will not give your the cinema dimming light experience.

What is interesting though in the area of multi-dimming is the fact that Plejd soon launches DAL-01. Typically each DALI bus can control 64 units, and DAL-01 seem to have one bus per unit. In theory, this is when multi-dimming becomes interesting and feasible with Plejd.

Update: 16 units->64 units, see [product manual](https://brand.plejd.com/downloads/manual/DAL-01/plejd_manual_DAL-01_SV.pdf).

According to the [product sheet](https://brand.plejd.com/downloads/product-sheet/DAL-01/plejd_product_sheet_DAL-01_SV_FI.pdf), it only supports _tunable white-belysning_ but the dali-alliance [certifcate](https://www.dali-alliance.org/products/3811/dal-01) indicates support for color control Tc. Roadmap maybe?

Anyway, room as light from Home Assistant will never be better than turning on a room from the Plejd app. The issue you describe in #152 is attributed to how Plejd works.

Short term: Transition units in areas and remove includeRoomsAsLights
Long term: Add support for DAL-01

Guess no-one wants to hear that, because DAL-01 will require new investments and adding a DALI bus (wires) in your installation.

But if you want the cinema dimming light experience, this is the way to go. And mind that the DALI ecosystem is evolving fast. Just search for Dali at eBay or Aliexpress to get a glimpse on where this is going. Or why not just jump over to [dali-alliance](https://www.dali-alliance.org/products) and have a look. >1700 formally certified products, and that does not include the Chinese ones...

I would love to get my hands on a DAL-01 from a beta tester to see how it signals on the BLE....
